merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/services/ConstraintAction.java

```diff
@@ -0,0 +1,244 @@
+      final var simStartTime = resultsHandle$
+          .map(gov.nasa.jpl.aerie.merlin.server.models.SimulationResultsHandle::startTime)
+          .orElse(plan.startTimestamp.toInstant());
+      final var simDuration = resultsHandle$
+          .map(SimulationResultsHandle::duration)
+          .orElse(Duration.of(
+              plan.startTimestamp.toInstant().until(plan.endTimestamp.toInstant(), ChronoUnit.MICROS),
+              Duration.MICROSECONDS));
+      final var simOffset = Duration.of(
+          plan.startTimestamp.toInstant().until(simStartTime, ChronoUnit.MICROS),
+          Duration.MICROSECONDS);
+
+      final var activities = new ArrayList<ActivityInstance>();
+      final var simulatedActivities = resultsHandle$
+          .map(SimulationResultsHandle::getSimulatedActivities)
+          .orElseGet(Collections::emptyMap);
+      for (final var entry : simulatedActivities.entrySet()) {
+        final var id = entry.getKey();
+        final var activity = entry.getValue();
+
+        final var activityOffset = Duration.of(
+            simStartTime.until(activity.start(), ChronoUnit.MICROS),
+            Duration.MICROSECONDS);
+
+        activities.add(new ActivityInstance(
+            id.id(),
+            activity.type(),
+            activity.arguments(),
+            Interval.between(activityOffset, activityOffset.plus(activity.duration()))));
+      }
+
+      final var externalDatasets = this.planService.getExternalDatasets(planId, simulationDatasetId);
+      final var realExternalProfiles = new HashMap<String, LinearProfile>();
+      final var discreteExternalProfiles = new HashMap<String, DiscreteProfile>();
+
+      for (final var pair : externalDatasets) {
+        final var offsetFromSimulationStart = pair.getLeft().minus(simOffset);
+        final var profileSet = pair.getRight();
+
+        for (final var profile : profileSet.discreteProfiles().entrySet()) {
+          discreteExternalProfiles.put(
+              profile.getKey(),
+              DiscreteProfile.fromExternalProfile(
+                  offsetFromSimulationStart,
+                  profile.getValue().getRight()));
+        }
+        for (final var profile : profileSet.realProfiles().entrySet()) {
+          realExternalProfiles.put(
+              profile.getKey(),
+              LinearProfile.fromExternalProfile(
+                  offsetFromSimulationStart,
+                  profile.getValue().getRight()));
+        }
+      }
+
+      final var environment = new EvaluationEnvironment(realExternalProfiles, discreteExternalProfiles);
+
+      final var realProfiles = new HashMap<String, LinearProfile>();
+      final var discreteProfiles = new HashMap<String, DiscreteProfile>();
+
+      for (final var entry : constraintCode.entrySet()) {
+
+        // Pipeline switch
+        // To remove the old constraints pipeline, delete the `useNewConstraintPipeline` variable
+        // and the else branch of this if statement.
```

Comment:
What is this comment referring to?

Reply:
I'm not exactly sure, more comments that were there when I got to the code.

Reply:
I'd argue to remove the comment then, since the code it's on doesn't relate to the comment.